manifests/server.pp

```diff
@@ -54,6 +54,19 @@
   $manage_pg_ident_conf       = $postgresql::params::manage_pg_ident_conf,
   $manage_recovery_conf       = $postgresql::params::manage_recovery_conf,
 
+  $config_entries  = {},
+  $databases       = {},
+  $database_grants = {},
+  $dbs             = {},
+  $extensions      = {},
+  $pg_hba_rules    = {},
+  $pg_ident_rules  = {},
+  $recovery        = {},
+  $roles           = {},
+  $schemas         = {},
+  $table_grants    = {},
+  $tablespaces     = {},
+
```

Comment:
Instead of adding a lot of extra hash parameters that are implicitly for hiera or ENC use, what about making this a bit more explicit? eg:

```puppet
class postgresql::profile::hiera (
  $config_entries  = {},
  $databases       = {},
  $database_grants = {},
  $dbs             = {},
  $extensions      = {},
  $pg_hba_rules    = {},
  $pg_ident_rules  = {},
  $recovery        = {},
  $roles           = {},
  $schemas         = {},
  $table_grants    = {},
  $tablespaces     = {},
) {
  create_resources('postgresql::server::config_entry',   $config_entries)
  create_resources('postgresql::server::database',       $databases)
  create_resources('postgresql::server::database_grant', $database_grants)
  create_resources('postgresql::server::db',             $dbs)
  create_resources('postgresql::server::extension',      $extensions)
  create_resources('postgresql::server::pg_hba_rule',    $pg_hba_rules)
  create_resources('postgresql::server::pg_ident_rule',  $pg_ident_rules)
  create_resources('postgresql::server::recovery',       $recovery)
  create_resources('postgresql::server::role',           $roles)
  create_resources('postgresql::server::schema',         $schemas)
  create_resources('postgresql::server::table_grant',    $table_grants)
  create_resources('postgresql::server::tablespace',     $tablespaces)
}
```